lldb/include/lldb/DataFormatters/TypeSummary.h

```diff
@@ -258,6 +258,14 @@ class TypeSummaryImpl {
 
   virtual std::string GetDescription() = 0;
 
+  /// Get the name of the Type Summary Provider, either a C++ class, a summary
+  /// string, or a script function name.
+  virtual std::string GetName() = 0;
+
+  /// Get the name of the kind of Summary Provider, either c++, summary string,
+  /// script or python.
+  virtual std::string GetSummaryKindName() = 0;
```

Comment:
Should this be a string or an enum? I would vote for an `lldb_internal` enum. That way plug-ins know what to choose form and don't make up random stuff.

Reply:
I'm not opposed to the enum, but I think if we're using an enum we should leverage 'TypeSummaryKind`, the existing kind enum and we should to string that.

My question is how we handle script, currently I check if it's a `.py` and if it is we call it Python. Do we want to default to Python for all scripts?

@Michael137 I talked to Greg offline about this, but do we support any SWIG interfaced languages other than Python? I believe I've seen Lua in the wild. If that's the case I'd like to keep the differentiation, but for this first patch we could get away with just `Python`